src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -183,15 +195,10 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
     // point when we try to access it. Calling this method may cause this
     // InteractiveWindow object to dispose itself
     private async tryGetMatchingNotebookDocument(): Promise<NotebookDocument | undefined> {
-        await this.interactiveOpenPromise;
-
         const notebookDocument = workspace.notebookDocuments.find(
-            (document) => this._notebookUri?.toString() === document.uri.toString()
+            (document) => this.notebookDocument.uri.toString() === document.uri.toString()
```

Comment:
```suggestion
            (document) => this.notebookDocument === document
```

Reply:
This doesn't make sense, I can't understand how we'd have the object reference yet not have the notebook in the list of `workspace.notebookDocuments`.

Can't we just check `this.notebookDocument.isClosed` in this method and return undefined.
& optionally add an event handler to the `workspace.onDidCloseNotebookDocument(() => this.dispose())` (if the same notebook has been closed).